test/acl_auto_configure_test.cpp

```diff
@@ -261,6 +269,27 @@ TEST(auto_configure, simple) {
               (int)m_device_def.autodiscovery_def.accel[0].max_work_group_size);
   CHECK_EQUAL(1, (int)m_device_def.autodiscovery_def.accel[0].is_sycl_compile);
 
+  // Checks for device global entry.
+  CHECK_EQUAL(2, m_device_def.autodiscovery_def.num_device_global);
+  CHECK(m_device_def.autodiscovery_def.device_global_mem_defs.find(
+            "kernel15_dev_global") !=
+        m_device_def.autodiscovery_def.device_global_mem_defs.end());
```

Comment:
You can store the iterator:
```suggestion
  const auto kernel15_dev_global = m_device_def.autodiscovery_def.device_global_mem_defs.find(
            "kernel15_dev_global");
  CHECK(kernel15_dev_global !=
        m_device_def.autodiscovery_def.device_global_mem_defs.end());
```